src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSPackageRestoreManager.cs

```diff
@@ -35,9 +35,6 @@ internal sealed class VSPackageRestoreManager : PackageRestoreManager
 
         private void OnSolutionOpenedOrClosed(object sender, EventArgs e)
         {
-            // This is a solution event. Should be on the UI thread
-            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
I'll be very cautious here to remove this line since it runs raisemissingpackages afterwards which uses SolutionManager alot which ultimately needs UI thread. Have you tested this code when PMC is already opened, you load a solution which is missing some packages... It should not hang or throw some weird errors in that case.